### ISSUE TITLE
As an administrator I should be able to install passbolt on a domain without a TLD.

### ISSUE THREAD

**Issue by stripthis** [passbolt/passbolt_api#238]
Ref. https://github.com/passbolt/passbolt_api/issues/55#issuecomment-379159618

Passbolt expect a FQDN with a TLD for the domain url. This breaks otherwise valid internal LAN FQDNs that do not end in .com, .net, .local, and whatever others (even though an internal DNS fully resolves the hostname, and SSL common name matches), such as https://passbolt

Solution: make the TLD optional in the url check for the domain on both server and client.

**Comment by alexdonh** [comment]
Anybody's having a look into this one?

**Comment by stripthis** [comment]
@alexdonh not really, this is not high enough on the priority list right now.

**Comment by alexdonh** [comment]
Well, I did a very quick look into the code. Looks like who's going to do this probably needs to have a custom `Validation` class from `Cake\Validation\Validation` to overwrite the `hostname` pattern:

```
    protected static $_pattern = [
        'hostname' => '(?:[_\p{L}0-9][-_\p{L}0-9]*\.)*(?:[\p{L}0-9][-\p{L}0-9]{0,62})\.(?:(?:[a-z]{2}\.)?[a-z]{2,})',
        'latitude' => '[-+]?([1-8]?\d(\.\d+)?|90(\.0+)?)',
        'longitude' => '[-+]?(180(\.0+)?|((1[0-7]\d)|([1-9]?\d))(\.\d+)?)',
    ];
```

URLs and Emails are being validated against this rule, which obviously fails on ones without proper TLDs.

That's for the server only. Client extensions needs to be fixed too. One of the solution could be passing `require_tld: false` to the 2nd parameter when validating URL (using `Validator.isURL`) and/or email (using `Validator.isEmail`).

For instance, in `background_page/model/userSettings.js`:

```
UserSettings.prototype.validateDomain = function (domain) {
  if ((typeof domain === 'undefined' || domain === '')) {
    throw new Error(__('A domain cannot be empty'));
  }
  if (!Validator.isURL(domain, { require_tld: false })) {
    throw new Error(__('The trusted domain url is not valid.'));
  }
  return true;
};
```

I can't manage to have some spare time to work on this and test it properly so... sorry. Hope this helps anyway.

**Comment by stripthis** [comment]
Yes that's pretty much where we are. Also I would like this require_tld check to be enabled by default and disabled by server side configuration, e.g. the same way we did with ```Configure::read('passbolt.email.validate.mx')```.

**Comment by pedrobrost** [comment]
any news about this?

**Comment by stripthis** [comment]
@pedrobrost no news unfortunately it is not a priority at the moment.

**Comment by sgargel** [comment]
I'm pretty sure u can use name like `https://passbolt.mynotvalidtld`

**Comment by RedOrion** [comment]
@sgargel That didn't work for me as of latest release as noted on [https://community.passbolt.com/t/error-something-unexpected-happened-the-setup-cannot-be-completed/1228/5](forums)

**Comment by stripthis** [comment]
Fixed with v2.13
